### PR TITLE
cupy: add remote error to the set of errors from nccl

### PR DESCRIPTION
**Pull request by mohitreddy1996**
Error codes possible from nccl - https://github.com/NVIDIA/nccl/blob/2ea4ee94bfb04c886c79ccae60ac9961000fdee2/src/nccl.h.in#L43

Fixes/Returns a more reasonable error for #8661

### CODE REVIEW

File: cupy_backends/cuda/libs/nccl.pyx [hunk 1]
```diff
@@ -98,6 +98,7 @@ cdef dict ERROR2 = {
     3: 'NCCL_ERROR_INTERNAL_ERROR',
     4: 'NCCL_ERROR_INVALID_ARGUMENT',
     5: 'NCCL_ERROR_INVALID_USAGE',
+    6: 'NCCL_ERROR_REMOTE_ERROR'
```
Comment (kmaehashi) [inline cupy_backends/cuda/libs/nccl.pyx:101]:
```suggestion
    6: 'NCCL_ERROR_REMOTE_ERROR',
```

**Comment by kmaehashi** [comment]
Thanks, LGTM!

/test mini

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
Test failures are unrealted. Thanks!

**Comment by mohitreddy1996** [comment]
Thank you @kmaehashi :D

**Pull request merged**